ArduCopter/control_sport.cpp

```diff
@@ -24,37 +24,28 @@ bool Copter::sport_init(bool ignore_checks)
 // should be called at 100hz or more
 void Copter::sport_run()
 {
-    float target_roll_rate, target_pitch_rate, target_yaw_rate;
-    float target_climb_rate = 0;
+    SportModeState sport_state;
     float takeoff_climb_rate = 0.0f;
 
     // initialize vertical speed and acceleration
     pos_control.set_speed_z(-g.pilot_velocity_z_max, g.pilot_velocity_z_max);
     pos_control.set_accel_z(g.pilot_accel_z);
 
-    // if not armed or throttle at zero, set throttle to zero and exit immediately
-    if (!motors.armed() || ap.throttle_zero || !motors.get_interlock()) {
-        motors.set_desired_spool_state(AP_Motors::DESIRED_SPIN_WHEN_ARMED);
-        attitude_control.set_throttle_out_unstabilized(0,true,g.throttle_filt);
-        pos_control.relax_alt_hold_controllers(get_throttle_pre_takeoff(channel_throttle->get_control_in())-motors.get_throttle_hover());
-        return;
-    }
-
     // apply SIMPLE mode transform
     update_simple_mode();
 
     // get pilot's desired roll and pitch rates
 
     // calculate rate requests
-    target_roll_rate = channel_roll->get_control_in() * g.acro_rp_p;
-    target_pitch_rate = channel_pitch->get_control_in() * g.acro_rp_p;
+    float target_roll_rate = channel_roll->get_control_in() * g.acro_rp_p;
+    float target_pitch_rate = channel_pitch->get_control_in() * g.acro_rp_p;
 
     int32_t roll_angle = wrap_180_cd(ahrs.roll_sensor);
-    target_roll_rate -= constrain_int32(roll_angle, -ACRO_LEVEL_MAX_ANGLE, ACRO_LEVEL_MAX_ANGLE) * g.acro_balance_roll;
+    target_roll_rate -= constrain_float(roll_angle, -ACRO_LEVEL_MAX_ANGLE, ACRO_LEVEL_MAX_ANGLE) * g.acro_balance_roll;
```

Comment:
I thought that because target_roll_rate was a float the constraint should also be a float..?  I'll revert it to an int32_t.
I guess there's a good question as to why some variables are floats and others are int32_t.  I don't think there's a good reason for that really.  We've been tending towards making everything a float.